talk/objectorientation/functors.tex

```diff
@@ -36,13 +36,13 @@
     struct BinaryFunction {
       virtual double operator() (double a, double b) = 0;
     };
-    struct Add : public BinaryFunction {
-      double operator() (double a, double b) override
-      { return a+b; }
-    };
     double binary_op(double a, double b, BinaryFunction &func) {
       return func(a, b);
     }
+    struct Add : BinaryFunction {
+      double operator() (double a, double b) override
+      { return a+b; }
+    };
     Add addfunc;
     double c = binary_op(a, b, addfunc);
```

Comment:
I found it more logical to show that `binary_op` can be implemented without knowing `Add` yet.